classes/plugins/ThemePlugin.inc.php

```diff
@@ -877,6 +881,146 @@ public function isColourDark($color, $limit = 130)
         );
         return $contrast < $limit;
     }
+
+    /**
+     * Add usage statistics graph to submission view page
+     */
+    public function displayUsageStatsGraph(int $submissionId): void
+    {
+        $this->addUsageStatsJavascriptData($this->getAllDownloadsStats($submissionId), $submissionId);
+        $this->loadChartJavascript();
+    }
+
+    /**
+     * Get the context for inclusion of usage stats display related JavaScripts in the submission view page
+     */
+    abstract protected function getSubmissionViewContext(): string;
+
+    /**
+     * Add submission's monthly statistics data to the script data output for graph display
+     */
+    protected function addUsageStatsJavascriptData(array $statsByMonth, int $submissionId): void
+    {
+        // Initialize the name space
+        $script_data = 'var pkpUsageStats = pkpUsageStats || {};';
+        $script_data .= 'pkpUsageStats.data = pkpUsageStats.data || {};';
+        $script_data .= 'pkpUsageStats.data.Submission = pkpUsageStats.data.Submission || {};';
+        $namespace = 'Submission[' . $submissionId . ']';
+        $script_data .= 'pkpUsageStats.data.' . $namespace . ' = ' . json_encode($statsByMonth) . ';';
+
+        $request = Application::get()->getRequest();
+        $templateMgr = TemplateManager::getManager($request);
+        $templateMgr->addJavaScript(
+            'pkpUsageStatsData',
+            $script_data,
+            [
+                'inline' => true,
+                'contexts' => $this->getSubmissionViewContext(),
+            ]
+        );
+    }
+
+    /**
+     * Load JavaScript assets for usage statistics display and pass data to the scripts
+     */
+    protected function loadChartJavascript(): void
+    {
+        $request = Application::get()->getRequest();
+        $templateMgr = TemplateManager::getManager($request);
+
+        // Register Chart.js on the frontend article view
+        $templateMgr->addJavaScript(
+            'chartJS',
+            'https://cdnjs.cloudflare.com/ajax/libs/Chart.js/2.0.1/Chart.js',
```

Comment:
We used to have an `enable_cdn` setting in `config.inc.php` that selected whether the software used a CDN or a local copy of the JS. That was removed with https://github.com/pkp/pkp-lib/issues/6478 and we're now using local copies of things (as far as I know). This effectively hides a JS dependency where it's hard to find in the codebase; I'd recommend adding it to our npm package set and linking to it locally.